ingestion_server/ingestion_server/es_mapping.py

```diff
@@ -137,6 +137,7 @@ def index_settings(table_name):
             "sample_rate": {"type": "integer"},
             "genres": {"fields": {"keyword": {"type": "keyword"}}, "type": "text"},
             "duration": {"type": "integer"},
+            "length": {"fields": {"keyword": {"type": "keyword"}}, "type": "text"},
```

Comment:
My confusion comes down to this: it seems like we were previously putting a string (from `get_duration`) into an integer field in ES. Did ES not complain about that before? Or am I misunderstanding the code?

I'm also not personally a fan of having length and duration in the index, especially if `length` was getting mapped to the string field before. It'll be pretty confusing.

I think we should "live with" the current name "duration" if it's getting mapped to the string field and correct the type, maybe all we need to make this work (in addition to adding the medium option to the enum, is to correct the type?).

I don't understand what the purpose of the name change is, basically.

Reply:
ES didn't complain because we weren't even calling the `get_duration` function . We did have it in the codebase, but we were putting the actual integer value into the index. And since the `duration` wasn't in the search query parameters list, we didn't even notice it.

Reply:
Ahh, I see. Because it wasn't I called I assumed there was some magic going on, like with Django serializer method fields :upside_down_face: That's the worst part of magic to me, it makes you start expecting it in places where it doesn't exist :cry: 

So then I'm still left wondering if pushing the enum into the index is the right way to do this over say, using a [range query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-range-query.html)? If our definitions of short/medium/long changed, for example, then we'd have to update every audio in the index. Or if we wanted to add an `extra_long` or `extra_short` for example, we'd have to update all the existing `short` or `long` to further discriminate them into the more specific groups.

If we use the range query, then it's just a matter of updating the query configuration against the existing `duration` integer field.

This may also apply to image dimensions or even aspect ratio (if we changed the aspect ratio to a decimal number instead of the enum).

Doing this for the image fields would require re-indexing, we might just have to live with the current design for the image fields. But we don't have to make the same problem for the audio duration (if it is indeed a problem).

Reply:
I think it was implemented as an Enum because the image `size` was implemented this way :)
The `range query` looks much more sustainable in the long run, for the reasons you describe. Now I'd have to find out how to implement it with Django.

Reply:
I wonder how expensive re-indexing images (replacing Enum size with the range query) would be?

Reply:
My guess is "very" but it's an uninformed one :slightly_smiling_face: